common/hyperscaler/rules/rules_service.go

```diff
@@ -23,5 +32,104 @@ func NewRulesServiceFromFile(rulesFilePath string) (*RulesService, error) {
 		return nil, err
 	}
 
-	return &RulesService{}, nil
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, err
+}
+
+func NewRulesServiceFromString(rules string, sort, unique, signature bool) (*RulesService, error) {
+	entries := strings.Split(rules, ";")
+
+	rulesConfig := &RulesConfig{
+		Rules: entries,
+	}
+
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, nil
+}
+
+func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
+	results := NewParsingResults()
+
+	for _, entry := range rulesConfig.Rules {
+		rule, err := rs.parser.Parse(entry)
+
+		results.Apply(entry, rule, err)
+	}
+
+	if rs.sort {
+		results.Results = SortRuleEntries(results.Results)
+	}
+
+	if rs.unique {
```

Comment:
We have here parsing, checking and sorting.
These responsibilities could be separated.

Reply:
There are only two clients of that method. One is KEB's main.go where we will invoke all of those actions in one place. The second is cli where we selectively turn on and off specific stages of processing. That is why this is a method to glues the process steps together.

Reply:
That is why I suggested to remove possibility of switching these checks/functions separately.
User needs to distinguish these concepts and use proper switches.
But the most important thing if file is ok and KEB will start with such config.
So by default checking should be the most restrictive.

Reply:
Even if there are only two clients I would vote for separation.
Or for name change, because this is not only parsing.

Reply:
Feel free to change however you wish.